### ISSUE TITLE
[Dynamic Region] DAG response is too big, please check config about region size or region merge scheduler

### ISSUE THREAD

**Issue by lilinghai** [pingcap/tiflash#7542]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

ch workload 1000 warehouse
query
```sql
select
    s_nationkey as supp_nation,
    substr(c_state, 1, 1) as cust_nation,
    extract(
        year
        from
            o_entry_d
    ) as l_year,
    sum(ol_amount) as revenue
from
    supplier,
    stock,
    order_line,
    orders,
    customer,
    nation n1,
    nation n2
where
    ol_supply_w_id = s_w_id
    and ol_i_id = s_i_id
    and mod((s_w_id * s_i_id), 10000) = s_suppkey
    and ol_w_id = o_w_id
    and ol_d_id = o_d_id
    and ol_o_id = o_id
    and c_id = o_c_id
    and c_w_id = o_w_id
    and c_d_id = o_d_id
    and s_nationkey = n1.n_nationkey
    and ascii(substr(c_state, 1, 1)) - 65 = n2.n_nationkey
    and (
        (
            n1.n_name = 'JAPAN'
            and n2.n_name = 'CHINA'
        )
        or (
            n1.n_name = 'CHINA'
            and n2.n_name = 'JAPAN'
        )
    )
    and ol_delivery_d between '2007-01-02 00:00:00.000000'
    and '2032-01-02 00:00:00.000000'
group by
    s_nationkey,
    substr(c_state, 1, 1),
    extract(
        year
        from
            o_entry_d
    )
order by
    s_nationkey,
    cust_nation,
    l_year;
```
### 2. What did you expect to see? (Required)
success
### 3. What did you see instead (Required)
query failed
 ERROR 1105 (HY000): other error for mpp stream: From MPP<query:<query_ts:1684917580465854823, local_query_id:20, server_id:3182238, start_ts:441691034196377601>,task_id:8>: Code: 0, e.displayText() = DB::Exception: code: 13
msg: "[FLASH:Coprocessor:Internal] DAG response is too big, please check config about region size or region merge scheduler"
, e.what() = DB::Exception,

### 4. What is your TiFlash version? (Required)
master
[2023/05/24 16:01:45.154 +08:00] [INFO] [client.go:509] ["Cluster version information"] [type=tidb] [version=7.2.0-alpha] [git_hash=909ff3c7ea42982a6bbe6cc5c4a017d4817a507e]
[2023/05/24 16:01:45.154 +08:00] [INFO] [client.go:509] ["Cluster version information"] [type=tikv] [version=7.2.0-alpha] [git_hash=46a09a42ccc831f27b18dcc2150d1fbf1304cb01]
[2023/05/24 16:01:45.154 +08:00] [INFO] [client.go:509] ["Cluster version information"] [type=pd] [version=7.2.0-alpha] [git_hash=8eddef57e9226a6aaf82eb9ae0878b6ca0ccfdd4]
[2023/05/24 16:01:45.154 +08:00] [INFO] [client.go:509] ["Cluster version information"] [type=tiflash] [version=7.2.0-alpha] [git_hash=c1d2b83efe97cc106ba095dbd6ac8146de29552e]

**Comment by CalvinNeo** [comment]
There are two many regions that exceeds 2GB
```
>>> [x['approximate_size'] for x in j['regions']]
[1, 1, 2946, 3123, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 4772, 1, 1, 1, 1, 1, 2374, 6, 1, 1, 1, 1, 1, 1, 2710, 1, 1, 1, 1, 1, 1, 8925, 1, 1, 1, 1, 582, 1, 8430, 1, 1, 1, 2296, 1, 1, 1, 726, 11882, 1, 1, 1, 1, 1, 3338, 1, 1, 1, 12640, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 12671, 6, 1, 1, 1, 1, 1, 1, 4653, 11183, 1, 1, 1, 3955, 1, 8962, 12036, 1918, 1, 1, 1, 12000, 1916, 1, 1, 1, 12030, 8665, 1, 1]
```

**Comment by windtalker** [comment]
When `Dynamic Region` is on, is it expected to have regions that exceeds 2GB? I think the cop request is not generated by TiDB, it should be caused by remote read. So if regions exceeds 2GB is expected, the remote read should be enhanced to be compatiable with `Dynamic Region`

**Comment by CalvinNeo** [comment]
This is not a ra storage issue, so we will have it closed.

**Comment by windtalker** [comment]
This issue should be fixed by #7805
